src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -20,64 +23,97 @@ trait ClusterComponent extends LeoComponent {
   import profile.api._
 
   class ClusterTable(tag: Tag) extends Table[ClusterRecord](tag, "CLUSTER") {
-    def clusterId =             column[UUID]              ("clusterId",             O.PrimaryKey)
+    def id =                    column[Long]              ("id",                    O.PrimaryKey, O.AutoInc)
     def clusterName =           column[String]            ("clusterName",           O.Length(254))
+    def googleId =              column[UUID]              ("googleId")
     def googleProject =         column[String]            ("googleProject",         O.Length(254))
     def googleServiceAccount =  column[String]            ("googleServiceAccount",  O.Length(254))
     def googleBucket =          column[String]            ("googleBucket",          O.Length(254))
     def operationName =         column[String]            ("operationName",         O.Length(254))
+    def status =                column[String]            ("status",                O.Length(254))
+    def hostIp =                column[Option[String]]    ("hostIp",                O.Length(254))
     def createdDate =           column[Timestamp]         ("createdDate",           O.SqlType("TIMESTAMP(6)"))
     def destroyedDate =         column[Option[Timestamp]] ("destroyedDate",         O.SqlType("TIMESTAMP(6)"))
 
-    def * = (clusterId, clusterName, googleProject, googleServiceAccount, googleBucket, operationName, createdDate, destroyedDate) <> (ClusterRecord.tupled, ClusterRecord.unapply)
+    def * = (id, clusterName, googleId, googleProject, googleServiceAccount, googleBucket, operationName, status, hostIp, createdDate, destroyedDate) <> (ClusterRecord.tupled, ClusterRecord.unapply)
   }
 
   object clusterQuery extends TableQuery(new ClusterTable(_)) {
 
     def save(cluster: Cluster): DBIO[Cluster] = {
-      (clusterQuery += marshalCluster(cluster)) flatMap { _ =>
-        labelQuery.saveAll(cluster.clusterId, cluster.labels)
+      (clusterQuery returning clusterQuery.map(_.id) += marshalCluster(cluster)) flatMap { clusterId =>
+        labelQuery.saveAll(clusterId, cluster.labels)
       } map { _ => cluster }
     }
 
     def list(): DBIO[Seq[Cluster]] = {
       clusterQuery.result flatMap { recs =>
-        val clusterActions = recs map { rec =>
-          labelQuery.getAll(rec.clusterId) map { labels =>
-            unmarshalCluster(rec, labels)
-          }
-        }
-        DBIO.sequence(clusterActions)
+        DBIO.sequence(recs map unmarshalWithLabels)
       }
     }
 
-    def delete(id: UUID): DBIO[Int] = {
+    def getByName(name: String): DBIO[Option[Cluster]] = {
```

Comment:
project + name should be unique here, not just name (and there should be a unique idx + constraint in the db if there isn't)